File: app/src/main/java/com/perflyst/twire/views/recyclerviews/auto_span_behaviours/AutoSpanBehaviour.java
package com.perflyst.twire.views.recyclerviews.auto_span_behaviours;

import android.content.Context;

import com.perflyst.twire.service.Settings;

/**
 * Created by Sebastian Rask on 09-05-2017.
 */

public interface AutoSpanBehaviour {
    String getElementSizeName(Settings settings);

    int getElementWidth(Context context);
}

File: app/src/main/java/com/perflyst/twire/views/recyclerviews/auto_span_behaviours/AutoSpanBehaviour.kt
package com.perflyst.twire.views.recyclerviews.auto_span_behaviours

import android.content.Context

/**
 * Created by Sebastian Rask on 09-05-2017.
 */
interface AutoSpanBehaviour {
    val elementSizeName: String

    fun getElementWidth(context: Context): Int
}
